zaza/openstack/utilities/openstack.py

```diff
@@ -2695,6 +2740,8 @@ def create_image(glance, image_url, image_name, image_cache_dir=None, tags=[],
         logging.info('Cached image found at {} - Skipping download'.format(
             local_path))
 
+    local_path = _convert_image_if_needed(local_path, disk_format)
+
```

Comment:
Unfortunately, this breaks the contract of the function's signature.  The `disk_format='qcow2'` as a default is no longer true, and this will silently nullify (for example), `test_412_image_conversion()` in zaza-openstack-tests that tests whether server-side image conversion works.

By extension it also breaks `add_image()` as that also has a `disk_format='qcow2'` default parameter, and even if the caller asks specifically for `qcow2`, they can't get it now.

So unfortunately, this is the wrong place to do this image conversion.

Reply:
The problem I see now is that we have a contradiction on the reason for using uploading qcow2 images or raw images.

If we are using ceph, we should be using raw images to prevent major impact to the testing time, which is the motivation of this PR.
On the other hand, if we need to test image conversion, then we need at least a qcow2 image uploaded instead of only raw images.

I understand the contract violation, I am still thinking about how to adjust it, but I would like to propose starting from a different perspective. I believe the test "test_412_image_conversion" should upload a specific qcow2 image and test its conversion to raw, given that its what the test is really testing. By doing this, it could use a lighter image like cirros, and therefore not impact all other tests that are ALSO converting qcow2 to raw when they shouldn't need to.

Now, back to the contradiction, we remove the need to upload qcow2 if the backend is ceph (unless explicitely specified, such as the test 412 would be doing). We just need to figure out a clean function declaration parameters given it is a shared function. My priority is that the default behavior is the convertion if it is a ceph backend, so I am inclined along the lines of "force_specified_image_format=False" (needs a better name) parameter to be used only for test 412 or other situation where the image conversion needs to be overridden.

Reply:
Alex and I discussed in a meeting today and agreed to follow my proposal, where the new function parameter will be something like "convert_if_ceph=True", and only test 412 will specify convert_if_ceph=False to override.